### PR TITLE
unix,win: add UV_PIPE_NO_TRUNCATE flag

### PR DESCRIPTION
**Pull request by bnoordhuis**
Libuv's default behavior is to truncate long Unix socket paths. The flag tells it to return an error instead.

Fixes: https://github.com/libuv/libuv/issues/4036

### CODE REVIEW

File: docs/src/pipe.rst [hunk 1]
```diff
@@ -68,15 +68,16 @@ API
...
+    for unsupported flags without performing the bind operation.
 
     Supports Linux abstract namespace sockets. ``namelen`` must include
     the leading nul byte but not the trailing nul byte.
 
     .. note::
         Paths on Unix get truncated to ``sizeof(sockaddr_un.sun_path)`` bytes,
-        typically between 92 and 108 bytes.
+        typically between 92 and 108 bytes, unless the ``UV_PIPE_NO_TRUNCATE``
+        flag is specified, in which case an ``UV_EINVAL`` error is returned.
```
Comment (santigimeno) [inline docs/src/pipe.rst:80]: I think we missed adding `.. versionadded:: 1.46.0` to the new methods in the previous commit. We should add them here.

**Review by santigimeno — APPROVED** [review]

File: docs/src/pipe.rst (hunk 1 above)
Comment (bnoordhuis) [reply docs/src/pipe.rst:80]: Good point, I'll do that.

Are you okay with the flag's name? I don't love it but I couldn't think of anything better.

Comment (santigimeno) [reply docs/src/pipe.rst:80]: > Are you okay with the flag's name? I don't love it but I couldn't think of anything better.

My only problem with the is that's Unix only but we have some Windows flags on other methods so... Do we know what error windows returns when the size of the pipe name is larger than 256? If it's something we translate as `UV_EINVAL` we might be just name it `UV_PIPE_UNIX_NO_TRUNCATE` and make no changes on the Win side.

Comment (bnoordhuis) [reply docs/src/pipe.rst:80]: > Do we know what error windows returns when the size of the pipe name is larger than 256?

I'll try it on the CI and see what happens.

Comment (bnoordhuis) [reply docs/src/pipe.rst:80]: Something that translates to UV_EACCES. ERROR_NOACCESS or ERROR_CANT_ACCESS_FILE, most likely.

I'm inclined to keep the current behavior unchanged and let users opt in through the flag. Agree/disagree?

Comment (santigimeno) [reply docs/src/pipe.rst:80]: > I'm inclined to keep the current behavior unchanged and let users opt in through the flag. Agree/disagree?

Fair enough. :+1:

**Pull request merged**